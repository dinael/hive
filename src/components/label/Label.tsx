import { FC } from 'react'
import styles from './Label.module.scss'

export type LabelProps = {
  text: string
  required?: boolean
  isRequiredText?: string
  labelHidden?: boolean
  disabled?: boolean
  className?: string
  onlyText?: boolean
}

export const Label: FC<LabelProps> = ({
  text,
  onlyText = false,
  required = false,
  isRequiredText,
  labelHidden = false,
  disabled,
  className,
}: LabelProps) => {

  const Tag = `${onlyText ? 'span' : 'label'}` as keyof JSX.IntrinsicElements

  return (
    <Tag className={`
      ${styles['label-text-component']}
      ${styles[disabled ? 'is-disabled' : '']}
      ${styles[required ? 'is-required' : '']}
      ${labelHidden ? 'sr-only' : ''}
      ${className ? className : ''}
      `}>
      {text + ':'}
      {required ?
        <em
          className={styles['required-text']}
          title={isRequiredText? `${isRequiredText}` : 'is required'}>
          *
        </em>
        : null}
    </Tag>
  )
}

export default Label
