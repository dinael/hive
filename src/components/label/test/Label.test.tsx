import { describe, test, expect, vi } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/react'
import { Label, LabelProps } from '../Label'

describe('Label', ()=> {
  const defaultProps: LabelProps = {
    text: 'text-test',
    required: false,
    isRequiredText: 'isRequiredText-test',
    labelHidden: false,
    disabled: false,
    className: 'className-test',
  }

  test('renders snapshot', () => {
    const { container } = render(<Label {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

})
