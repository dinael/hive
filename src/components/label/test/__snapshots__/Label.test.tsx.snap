// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > Label renders snapshot 1`] = `
<div>
  <label
    class="
      label-text-component
      
      
      
      className-test"
  >
    text-test:
  </label>
</div>
`;

exports[`Label > Label renders with the correct tag name: label 1`] = `
<div>
  <label
    class="
      label-text-component
      
      
      
      className-test"
  >
    text-test:
  </label>
</div>
`;

exports[`Label > Label renders with the correct tag name: span 1`] = `
<div>
  <span
    class="
      label-text-component
      
      
      
      className-test"
  >
    text-test:
  </span>
</div>
`;
